### PR TITLE
X.U.Hacks: Add "Stacking trays (trayer) above panels (xmobar)"

### PR DESCRIPTION
**Pull request by liskin**
### Description

Placing `trayer` on top of `xmobar` is somewhat tricky:

- they both should be lowered to the bottom of the stacking order to avoid overlapping fullscreen windows

- the tray needs to be stacked on top of the panel regardless of which happens to start first

`trayerAboveXmobarEventHook` (and the more generic `trayAbovePanelEventHook`) is an event hook that ensures the latter: whenever the tray lowers itself to the bottom of the stack, it checks whether there are any panels above it and lowers these again.

### Checklist

  - [X] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

  - [X] I updated the `CHANGES.md` file

  - [X] I updated the `XMonad.Doc.Extending` file (if appropriate)

### CODE REVIEW

**Comment by liskin** [comment]
(I'm fully aware that using xmobar with `overrideRedirect = True` currently requires restarting it after every rescreen, and I intend to submit another PR for that soon.)

File: XMonad/Util/Hacks.hs [hunk 1]
```diff
@@ -92,3 +99,47 @@ javaHack conf = conf
...
+
+-- | Whenever a tray window lowers itself to the bottom of the stack, look for
+-- any panels above it and lower these.
+trayAbovePanelEventHook
+  :: Query Bool -- ^ tray
+  -> Query Bool -- ^ panel
+  -> (Event -> X All) -- ^ event hook
+trayAbovePanelEventHook trayQ panelQ ConfigureEvent{ev_window = w, ev_above = a} | a == none = do
+  -- when tray is lowered to the bottom of stack (a == none),
+  -- put all panels that are above it below
```
Comment (TheMC47) [inline XMonad/Util/Hacks.hs:137]: Maybe "move all panels below it"?

**Review by TheMC47 — APPROVED** [review]
LGTM

File: XMonad/Util/Hacks.hs (hunk 1 above)
Comment (liskin) [reply XMonad/Util/Hacks.hs:137]: Hm, do I need this comment at all? It's saying the same thing as the haddock for the function itself (which I wrote today and then forgot to take another look at the comment in code).

Comment (liskin) [reply XMonad/Util/Hacks.hs]: I dropped it. And then I added a usage to the the doc which I forgot.

**Pull request merged**